diagnostics-extension/src/controllers/diagnostics.ts

```diff
@@ -108,7 +112,7 @@ export const handleDiagnostics: DiagnosticsController = (req, res) => {
   switch (req.diagnostics.action) {
     case DiagnosticsAction.START:
       if (req.diagnostics.routineName === RoutineType.available_routines) {
-        return getAvailableRoutines();
+        return getAvailableRoutines(res);
```

Comment:
We should declare a new RoutineAction. Lets not embed `GetAvailableRoutine` inside `StartRoutine`.

Reply:
Another DiagnosticsAction value is added to deal with it.